> /scratch/07169/wgl/adjoint/python/Main_no_matrix.py(122)lossFn()
-> loss = jnp.abs(u[-1] - true)
> /scratch/07169/wgl/adjoint/python/Main_no_matrix.py(123)lossFn()
-> return loss
Traced<ShapedArray(float32[1])>with<JVPTrace(level=3/0)> with
  primal = Traced<ShapedArray(float32[1])>with<BatchTrace(level=1/0)> with
    val = Array([[0.95305663],
       [0.5101362 ],
       [0.8262037 ],
       [0.51679313],
       [0.61365414],
       [0.6466246 ],
       [0.08147666],
       [0.33450603],
       [0.4604448 ],
       [0.7013949 ],
       [0.19755459],
       [0.6748692 ],
       [0.2526596 ],
       [0.27847964],
       [0.17838943],
       [0.6246938 ],
       [0.16198133],
       [1.0274377 ],
       [0.58149636],
       [0.08797866],
       [0.68639743],
       [0.18528086],
       [0.6301379 ],
       [0.469609  ],
       [0.17276554],
       [0.62512267],
       [0.36342525],
       [0.5396408 ],
       [0.6802418 ],
       [0.3264718 ],
       [0.7365049 ],
       [0.5803627 ],
       [0.26177943],
       [0.4781177 ],
       [0.06888109],
       [0.41025245],
       [0.33878353],
       [0.5550431 ],
       [0.10958341],
       [0.3907553 ],
       [0.71874213],
       [0.5116054 ],
       [0.6411704 ],
       [0.6009389 ],
       [0.6337606 ],
       [0.7138085 ],
       [0.5677031 ],
       [0.5357355 ],
       [0.5658915 ],
       [0.35876644],
       [0.15635833],
       [0.55994177],
       [0.35313845],
       [0.71532273],
       [0.39576828],
       [1.5921162 ],
       [0.03949165],
       [0.69281894],
       [0.6338313 ],
       [0.21142632],
       [0.573329  ],
       [0.08086684],
       [0.55634725],
       [0.00236166],
       [0.36163718],
       [0.29836217],
       [0.7163925 ],
       [0.67771673],
       [0.7146686 ],
       [0.54061663],
       [0.6178097 ],
       [0.5611708 ],
       [0.69692856],
       [0.5788808 ],
       [0.37619972],
       [0.2707923 ],
       [0.5788189 ],
       [0.10761836],
       [0.605808  ],
       [0.6061494 ],
       [0.70974016],
       [0.6130805 ],
       [0.7057903 ],
       [0.52033687],
       [0.82201195],
       [0.2260156 ],
       [0.6045624 ],
       [0.39600202],
       [0.11008212],
       [0.53802437],
       [0.15790147],
       [0.6079054 ],
       [0.6445424 ],
       [0.669708  ],
       [0.4492061 ],
       [0.01295194],
       [1.487581  ],
       [0.63336486],
       [0.3675909 ],
       [0.54790294]], dtype=float32)
    batch_dim = 0
  tangent = Traced<ShapedArray(float32[1])>with<JaxprTrace(level=2/0)> with
    pval = (ShapedArray(float32[1]), None)
    recipe = JaxprEqnRecipe(eqn_id=<object object at 0x147e041bf800>, in_tracers=(Traced<ShapedArray(float32[1]):JaxprTrace(level=2/0)>, Traced<ShapedArray(bool[1]):JaxprTrace(level=2/0)>), out_tracer_refs=[<weakref at 0x147e041dfe00; to 'JaxprTracer' at 0x147e041df810>], out_avals=[ShapedArray(float32[1])], primitive=pjit, params={'jaxpr': { lambda ; a:f32[1] b:bool[1]. let
    c:f32[1] = neg a
    d:f32[1] = select_n b c a
  in (d,) }, 'in_shardings': (UnspecifiedValue, UnspecifiedValue), 'out_shardings': (UnspecifiedValue,), 'resource_env': None, 'donated_invars': (False, False), 'name': 'absolute', 'keep_unused': False, 'inline': True}, effects=set(), source_info=SourceInfo(traceback=<jaxlib.xla_extension.Traceback object at 0x147e04164430>, name_stack=NameStack(stack=(Transform(name='vmap'), Transform(name='jvp')))))
(1,)
Traced<ShapedArray(float32[])>with<BatchTrace(level=1/0)> with
  val = Array([-0.9901214 ,  0.64691544, -0.84815603, -0.50050396,  1.5895107 ,
        1.4664404 , -0.01347107,  2.6306264 ,  0.59137785,  1.1418931 ,
        0.29788852,  1.3237785 , -0.20471042,  0.3882548 , -0.12158161,
        1.5486751 ,  0.2582317 , -1.0739069 ,  1.7098631 , -0.02060837,
        0.8450341 , -0.12946618,  1.5278755 ,  0.60164165,  0.2701745 ,
        1.5463233 ,  2.522838  ,  1.8660336 ,  1.2910162 ,  2.6608582 ,
       -0.7472704 ,  0.72524947,  0.36963555,  2.0941403 ,  0.15432447,
        2.348291  , -0.30064887,  1.808442  ,  0.19968349,  0.5136955 ,
        0.9192349 ,  0.64849013,  1.4865669 ,  1.6369264 ,  1.5143209 ,
        0.98278993,  0.7110859 , -0.52135104,  1.7676992 ,  0.47799778,
       -0.09701091,  0.7024936 ,  0.4716296 ,  0.9635436 ,  0.5191753 ,
       -1.7006829 ,  3.732044  ,  1.2134037 ,  1.5141587 ,  0.31349415,
       -0.563759  ,  0.16783035,  1.8034276 ,  0.07484698,  0.48107812,
       -0.2555588 ,  0.94953144,  1.306865  ,  0.9715561 ,  1.8622403 ,
        0.76702565,  1.7851639 ,  1.1878061 ,  1.7189288 ,  2.4749691 ,
        0.37988564,  1.7194853 ,  0.1976695 ,  0.75363207,  0.7540112 ,
        1.035264  ,  1.5919094 ,  1.0863286 ,  0.6582892 , -0.8434679 ,
       -0.17471439,  1.6235855 ,  0.51946044, -0.04533577,  0.6779778 ,
        0.25376135,  1.6110857 ,  1.4744467 ,  1.3646972 ,  2.2028759 ,
        0.06304681, -1.5861652 , -0.6310212 ,  2.5073876 ,  0.6890312 ],      dtype=float32)
  batch_dim = 0
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/scratch/07169/wgl/adjoint/python/Main_no_matrix.py", line 186, in <module>
    params, opt_state = trainStep(u_0_train, t, dt, params, net, opt_state,
  File "/scratch/07169/wgl/adjoint/python/Main_no_matrix.py", line 130, in trainStep
    vmap(grad(lossFn, argnums=(3,)),
  File "/work/07169/wgl/ls6/PHO-ICES/adj-env/lib/python3.9/site-packages/jax/_src/traceback_util.py", line 166, in reraise_with_filtered_traceback
    return fun(*args, **kwargs)
  File "/work/07169/wgl/ls6/PHO-ICES/adj-env/lib/python3.9/site-packages/jax/_src/api.py", line 1253, in vmap_f
    out_flat = batching.batch(
  File "/work/07169/wgl/ls6/PHO-ICES/adj-env/lib/python3.9/site-packages/jax/_src/linear_util.py", line 166, in call_wrapped
    ans = self.f(*args, **dict(self.params, **kwargs))
  File "/work/07169/wgl/ls6/PHO-ICES/adj-env/lib/python3.9/site-packages/jax/_src/traceback_util.py", line 166, in reraise_with_filtered_traceback
    return fun(*args, **kwargs)
  File "/work/07169/wgl/ls6/PHO-ICES/adj-env/lib/python3.9/site-packages/jax/_src/api.py", line 659, in grad_f
    _, g = value_and_grad_f(*args, **kwargs)
  File "/work/07169/wgl/ls6/PHO-ICES/adj-env/lib/python3.9/site-packages/jax/_src/traceback_util.py", line 166, in reraise_with_filtered_traceback
    return fun(*args, **kwargs)
  File "/work/07169/wgl/ls6/PHO-ICES/adj-env/lib/python3.9/site-packages/jax/_src/api.py", line 739, in value_and_grad_f
    _check_scalar(ans)
  File "/work/07169/wgl/ls6/PHO-ICES/adj-env/lib/python3.9/site-packages/jax/_src/api.py", line 759, in _check_scalar
    raise TypeError(msg(f"had shape: {aval.shape}"))
jax._src.traceback_util.UnfilteredStackTrace: TypeError: Gradient only defined for scalar-output functions. Output had shape: (1,).
The stack trace below excludes JAX-internal frames.
The preceding is the original exception that occurred, unmodified.
--------------------
The above exception was the direct cause of the following exception:
Traceback (most recent call last):
  File "/opt/apps/intel19/python3/3.9.7/lib/python3.9/pdb.py", line 1723, in main
    pdb._runscript(mainpyfile)
  File "/opt/apps/intel19/python3/3.9.7/lib/python3.9/pdb.py", line 1583, in _runscript
    self.run(statement)
  File "/opt/apps/intel19/python3/3.9.7/lib/python3.9/bdb.py", line 580, in run
    exec(cmd, globals, locals)
  File "<string>", line 1, in <module>
  File "/scratch/07169/wgl/adjoint/python/Main_no_matrix.py", line 186, in <module>
    params, opt_state = trainStep(u_0_train, t, dt, params, net, opt_state,
  File "/scratch/07169/wgl/adjoint/python/Main_no_matrix.py", line 130, in trainStep
    vmap(grad(lossFn, argnums=(3,)),
TypeError: Gradient only defined for scalar-output functions. Output had shape: (1,).
Uncaught exception. Entering post mortem debugging
Running 'cont' or 'step' will restart the program
> /scratch/07169/wgl/adjoint/python/Main_no_matrix.py(130)trainStep()
-> vmap(grad(lossFn, argnums=(3,)),
Post mortem debugger finished. The /scratch/07169/wgl/adjoint/python/Main_no_matrix.py will be restarted
> /scratch/07169/wgl/adjoint/python/Main_no_matrix.py(1)<module>()
-> import os
> /scratch/07169/wgl/adjoint/python/Main_no_matrix.py(122)lossFn()
-> loss = jnp.abs(jnp.squeeze(u[-1]) - jnp.squeeze(true))
> /scratch/07169/wgl/adjoint/python/Main_no_matrix.py(122)lossFn()
-> loss = jnp.abs(jnp.squeeze(u[-1]) - jnp.squeeze(true))
Deleted breakpoint 1 at /scratch/07169/wgl/adjoint/python/Main_no_matrix.py:122
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/scratch/07169/wgl/adjoint/python/Main_no_matrix.py", line 200, in <module>
    v = adjointSolve(u, dt, ref_factor, params, net)
  File "/scratch/07169/wgl/adjoint/python/Main_no_matrix.py", line 60, in adjointSolve
    dJdU = grad(outFnl)(u_fine, t_fine)
  File "/work/07169/wgl/ls6/PHO-ICES/adj-env/lib/python3.9/site-packages/jax/_src/traceback_util.py", line 166, in reraise_with_filtered_traceback
    return fun(*args, **kwargs)
  File "/work/07169/wgl/ls6/PHO-ICES/adj-env/lib/python3.9/site-packages/jax/_src/api.py", line 659, in grad_f
    _, g = value_and_grad_f(*args, **kwargs)
  File "/work/07169/wgl/ls6/PHO-ICES/adj-env/lib/python3.9/site-packages/jax/_src/traceback_util.py", line 166, in reraise_with_filtered_traceback
    return fun(*args, **kwargs)
  File "/work/07169/wgl/ls6/PHO-ICES/adj-env/lib/python3.9/site-packages/jax/_src/api.py", line 735, in value_and_grad_f
    ans, vjp_py = _vjp(f_partial, *dyn_args, reduce_axes=reduce_axes)
  File "/work/07169/wgl/ls6/PHO-ICES/adj-env/lib/python3.9/site-packages/jax/_src/api.py", line 2258, in _vjp
    out_primal, out_vjp = ad.vjp(
  File "/work/07169/wgl/ls6/PHO-ICES/adj-env/lib/python3.9/site-packages/jax/_src/interpreters/ad.py", line 139, in vjp
    out_primals, pvals, jaxpr, consts = linearize(traceable, *primals)
  File "/work/07169/wgl/ls6/PHO-ICES/adj-env/lib/python3.9/site-packages/jax/_src/interpreters/ad.py", line 128, in linearize
    jaxpr, out_pvals, consts = pe.trace_to_jaxpr_nounits(jvpfun_flat, in_pvals)
  File "/work/07169/wgl/ls6/PHO-ICES/adj-env/lib/python3.9/site-packages/jax/_src/profiler.py", line 314, in wrapper
    return func(*args, **kwargs)
  File "/work/07169/wgl/ls6/PHO-ICES/adj-env/lib/python3.9/site-packages/jax/_src/interpreters/partial_eval.py", line 771, in trace_to_jaxpr_nounits
    jaxpr, (out_pvals, consts, env) = fun.call_wrapped(pvals)
  File "/work/07169/wgl/ls6/PHO-ICES/adj-env/lib/python3.9/site-packages/jax/_src/linear_util.py", line 166, in call_wrapped
    ans = self.f(*args, **dict(self.params, **kwargs))
  File "/scratch/07169/wgl/adjoint/python/Main_no_matrix.py", line 51, in outFnl
    loss = jnp.abs(u[-1], jnp.sin(t[-1]+u[0]))
  File "/work/07169/wgl/ls6/PHO-ICES/adj-env/lib/python3.9/site-packages/jax/_src/traceback_util.py", line 166, in reraise_with_filtered_traceback
    return fun(*args, **kwargs)
  File "/work/07169/wgl/ls6/PHO-ICES/adj-env/lib/python3.9/site-packages/jax/_src/pjit.py", line 238, in cache_miss
    outs, out_flat, out_tree, args_flat = _python_pjit_helper(
  File "/work/07169/wgl/ls6/PHO-ICES/adj-env/lib/python3.9/site-packages/jax/_src/pjit.py", line 180, in _python_pjit_helper
    args_flat, _, params, in_tree, out_tree, _ = infer_params_fn(
  File "/work/07169/wgl/ls6/PHO-ICES/adj-env/lib/python3.9/site-packages/jax/_src/api.py", line 311, in infer_params
    return pjit.common_infer_params(pjit_info_args, *args, **kwargs)
  File "/work/07169/wgl/ls6/PHO-ICES/adj-env/lib/python3.9/site-packages/jax/_src/pjit.py", line 480, in common_infer_params
    jaxpr, consts, canonicalized_out_shardings_flat = _pjit_jaxpr(
  File "/work/07169/wgl/ls6/PHO-ICES/adj-env/lib/python3.9/site-packages/jax/_src/pjit.py", line 918, in _pjit_jaxpr
    jaxpr, final_consts, out_type = _create_pjit_jaxpr(
  File "/work/07169/wgl/ls6/PHO-ICES/adj-env/lib/python3.9/site-packages/jax/_src/linear_util.py", line 322, in memoized_fun
    ans = call(fun, *args)
  File "/work/07169/wgl/ls6/PHO-ICES/adj-env/lib/python3.9/site-packages/jax/_src/pjit.py", line 874, in _create_pjit_jaxpr
    jaxpr, global_out_avals, consts = pe.trace_to_jaxpr_dynamic(
  File "/work/07169/wgl/ls6/PHO-ICES/adj-env/lib/python3.9/site-packages/jax/_src/profiler.py", line 314, in wrapper
    return func(*args, **kwargs)
  File "/work/07169/wgl/ls6/PHO-ICES/adj-env/lib/python3.9/site-packages/jax/_src/interpreters/partial_eval.py", line 2049, in trace_to_jaxpr_dynamic
    jaxpr, out_avals, consts = trace_to_subjaxpr_dynamic(
  File "/work/07169/wgl/ls6/PHO-ICES/adj-env/lib/python3.9/site-packages/jax/_src/interpreters/partial_eval.py", line 2066, in trace_to_subjaxpr_dynamic
    ans = fun.call_wrapped(*in_tracers_)
  File "/work/07169/wgl/ls6/PHO-ICES/adj-env/lib/python3.9/site-packages/jax/_src/linear_util.py", line 166, in call_wrapped
    ans = self.f(*args, **dict(self.params, **kwargs))
jax._src.traceback_util.UnfilteredStackTrace: TypeError: absolute() takes 1 positional argument but 2 were given
The stack trace below excludes JAX-internal frames.
The preceding is the original exception that occurred, unmodified.
--------------------
The above exception was the direct cause of the following exception:
Traceback (most recent call last):
  File "/opt/apps/intel19/python3/3.9.7/lib/python3.9/pdb.py", line 1723, in main
    pdb._runscript(mainpyfile)
  File "/opt/apps/intel19/python3/3.9.7/lib/python3.9/pdb.py", line 1583, in _runscript
    self.run(statement)
  File "/opt/apps/intel19/python3/3.9.7/lib/python3.9/bdb.py", line 580, in run
    exec(cmd, globals, locals)
  File "<string>", line 1, in <module>
  File "/scratch/07169/wgl/adjoint/python/Main_no_matrix.py", line 200, in <module>
    v = adjointSolve(u, dt, ref_factor, params, net)
  File "/scratch/07169/wgl/adjoint/python/Main_no_matrix.py", line 60, in adjointSolve
    dJdU = grad(outFnl)(u_fine, t_fine)
  File "/scratch/07169/wgl/adjoint/python/Main_no_matrix.py", line 51, in outFnl
    loss = jnp.abs(u[-1], jnp.sin(t[-1]+u[0]))
TypeError: absolute() takes 1 positional argument but 2 were given
Uncaught exception. Entering post mortem debugging
Running 'cont' or 'step' will restart the program
> /scratch/07169/wgl/adjoint/python/Main_no_matrix.py(51)outFnl()
-> loss = jnp.abs(u[-1], jnp.sin(t[-1]+u[0]))
Post mortem debugger finished. The /scratch/07169/wgl/adjoint/python/Main_no_matrix.py will be restarted
> /scratch/07169/wgl/adjoint/python/Main_no_matrix.py(1)<module>()
-> import os
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/scratch/07169/wgl/adjoint/python/Main_no_matrix.py", line 1, in <module>
    import os
  File "/scratch/07169/wgl/adjoint/python/Main_no_matrix.py", line 60, in adjointSolve
    dJdU = grad(outFnl)(u_fine, t_fine)
  File "/work/07169/wgl/ls6/PHO-ICES/adj-env/lib/python3.9/site-packages/jax/_src/traceback_util.py", line 166, in reraise_with_filtered_traceback
    return fun(*args, **kwargs)
  File "/work/07169/wgl/ls6/PHO-ICES/adj-env/lib/python3.9/site-packages/jax/_src/api.py", line 659, in grad_f
    _, g = value_and_grad_f(*args, **kwargs)
  File "/work/07169/wgl/ls6/PHO-ICES/adj-env/lib/python3.9/site-packages/jax/_src/traceback_util.py", line 166, in reraise_with_filtered_traceback
    return fun(*args, **kwargs)
  File "/work/07169/wgl/ls6/PHO-ICES/adj-env/lib/python3.9/site-packages/jax/_src/api.py", line 735, in value_and_grad_f
    ans, vjp_py = _vjp(f_partial, *dyn_args, reduce_axes=reduce_axes)
  File "/work/07169/wgl/ls6/PHO-ICES/adj-env/lib/python3.9/site-packages/jax/_src/api.py", line 2258, in _vjp
    out_primal, out_vjp = ad.vjp(
  File "/work/07169/wgl/ls6/PHO-ICES/adj-env/lib/python3.9/site-packages/jax/_src/interpreters/ad.py", line 139, in vjp
    out_primals, pvals, jaxpr, consts = linearize(traceable, *primals)
  File "/work/07169/wgl/ls6/PHO-ICES/adj-env/lib/python3.9/site-packages/jax/_src/interpreters/ad.py", line 128, in linearize
    jaxpr, out_pvals, consts = pe.trace_to_jaxpr_nounits(jvpfun_flat, in_pvals)
  File "/work/07169/wgl/ls6/PHO-ICES/adj-env/lib/python3.9/site-packages/jax/_src/profiler.py", line 314, in wrapper
    return func(*args, **kwargs)
  File "/work/07169/wgl/ls6/PHO-ICES/adj-env/lib/python3.9/site-packages/jax/_src/interpreters/partial_eval.py", line 771, in trace_to_jaxpr_nounits
    jaxpr, (out_pvals, consts, env) = fun.call_wrapped(pvals)
  File "/work/07169/wgl/ls6/PHO-ICES/adj-env/lib/python3.9/site-packages/jax/_src/linear_util.py", line 166, in call_wrapped
    ans = self.f(*args, **dict(self.params, **kwargs))
  File "/scratch/07169/wgl/adjoint/python/Main_no_matrix.py", line 51, in outFnl
    loss = jnp.abs(jnp.squeeze(u[-1]), jnp.squeezze(jnp.sin(t[-1] + u[0])))
  File "/work/07169/wgl/ls6/PHO-ICES/adj-env/lib/python3.9/site-packages/jax/_src/deprecations.py", line 53, in getattr
    raise AttributeError(f"module {module!r} has no attribute {name!r}")
jax._src.traceback_util.UnfilteredStackTrace: AttributeError: module 'jax.numpy' has no attribute 'squeezze'
The stack trace below excludes JAX-internal frames.
The preceding is the original exception that occurred, unmodified.
--------------------
The above exception was the direct cause of the following exception:
Traceback (most recent call last):
  File "/opt/apps/intel19/python3/3.9.7/lib/python3.9/pdb.py", line 1723, in main
    pdb._runscript(mainpyfile)
  File "/opt/apps/intel19/python3/3.9.7/lib/python3.9/pdb.py", line 1583, in _runscript
    self.run(statement)
  File "/opt/apps/intel19/python3/3.9.7/lib/python3.9/bdb.py", line 580, in run
    exec(cmd, globals, locals)
  File "<string>", line 1, in <module>
  File "/scratch/07169/wgl/adjoint/python/Main_no_matrix.py", line 1, in <module>
    import os
  File "/scratch/07169/wgl/adjoint/python/Main_no_matrix.py", line 60, in adjointSolve
    dJdU = grad(outFnl)(u_fine, t_fine)
  File "/scratch/07169/wgl/adjoint/python/Main_no_matrix.py", line 51, in outFnl
    loss = jnp.abs(jnp.squeeze(u[-1]), jnp.squeezze(jnp.sin(t[-1] + u[0])))
  File "/work/07169/wgl/ls6/PHO-ICES/adj-env/lib/python3.9/site-packages/jax/_src/deprecations.py", line 53, in getattr
    raise AttributeError(f"module {module!r} has no attribute {name!r}")
AttributeError: module 'jax.numpy' has no attribute 'squeezze'
Uncaught exception. Entering post mortem debugging
Running 'cont' or 'step' will restart the program
> /work/07169/wgl/ls6/PHO-ICES/adj-env/lib/python3.9/site-packages/jax/_src/deprecations.py(53)getattr()
-> raise AttributeError(f"module {module!r} has no attribute {name!r}")
Post mortem debugger finished. The /scratch/07169/wgl/adjoint/python/Main_no_matrix.py will be restarted
> /scratch/07169/wgl/adjoint/python/Main_no_matrix.py(1)<module>()
-> import os
Program interrupted. (Use 'cont' to resume).
--Return--
> /work/07169/wgl/ls6/PHO-ICES/adj-env/lib/python3.9/site-packages/numpy/core/numerictypes.py(420)issubdtype()->True
-> return issubclass(arg1, arg2)
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/scratch/07169/wgl/adjoint/python/Main_no_matrix.py", line 200, in <module>
    v = adjointSolve(u, dt, ref_factor, params, net)
  File "/scratch/07169/wgl/adjoint/python/Main_no_matrix.py", line 60, in adjointSolve
    dJdU = grad(outFnl)(u_fine, t_fine)
  File "/work/07169/wgl/ls6/PHO-ICES/adj-env/lib/python3.9/site-packages/jax/_src/traceback_util.py", line 166, in reraise_with_filtered_traceback
    return fun(*args, **kwargs)
  File "/work/07169/wgl/ls6/PHO-ICES/adj-env/lib/python3.9/site-packages/jax/_src/api.py", line 659, in grad_f
    _, g = value_and_grad_f(*args, **kwargs)
  File "/work/07169/wgl/ls6/PHO-ICES/adj-env/lib/python3.9/site-packages/jax/_src/traceback_util.py", line 166, in reraise_with_filtered_traceback
    return fun(*args, **kwargs)
  File "/work/07169/wgl/ls6/PHO-ICES/adj-env/lib/python3.9/site-packages/jax/_src/api.py", line 735, in value_and_grad_f
    ans, vjp_py = _vjp(f_partial, *dyn_args, reduce_axes=reduce_axes)
  File "/work/07169/wgl/ls6/PHO-ICES/adj-env/lib/python3.9/site-packages/jax/_src/api.py", line 2258, in _vjp
    out_primal, out_vjp = ad.vjp(
  File "/work/07169/wgl/ls6/PHO-ICES/adj-env/lib/python3.9/site-packages/jax/_src/interpreters/ad.py", line 139, in vjp
    out_primals, pvals, jaxpr, consts = linearize(traceable, *primals)
  File "/work/07169/wgl/ls6/PHO-ICES/adj-env/lib/python3.9/site-packages/jax/_src/interpreters/ad.py", line 128, in linearize
    jaxpr, out_pvals, consts = pe.trace_to_jaxpr_nounits(jvpfun_flat, in_pvals)
  File "/work/07169/wgl/ls6/PHO-ICES/adj-env/lib/python3.9/site-packages/jax/_src/profiler.py", line 314, in wrapper
    return func(*args, **kwargs)
  File "/work/07169/wgl/ls6/PHO-ICES/adj-env/lib/python3.9/site-packages/jax/_src/interpreters/partial_eval.py", line 771, in trace_to_jaxpr_nounits
    jaxpr, (out_pvals, consts, env) = fun.call_wrapped(pvals)
  File "/work/07169/wgl/ls6/PHO-ICES/adj-env/lib/python3.9/site-packages/jax/_src/linear_util.py", line 166, in call_wrapped
    ans = self.f(*args, **dict(self.params, **kwargs))
  File "/scratch/07169/wgl/adjoint/python/Main_no_matrix.py", line 51, in outFnl
    loss = jnp.abs(jnp.squeeze(u[-1]), jnp.squeeze(jnp.sin(t[-1] + u[0])))
  File "/work/07169/wgl/ls6/PHO-ICES/adj-env/lib/python3.9/site-packages/jax/_src/traceback_util.py", line 166, in reraise_with_filtered_traceback
    return fun(*args, **kwargs)
  File "/work/07169/wgl/ls6/PHO-ICES/adj-env/lib/python3.9/site-packages/jax/_src/pjit.py", line 238, in cache_miss
    outs, out_flat, out_tree, args_flat = _python_pjit_helper(
  File "/work/07169/wgl/ls6/PHO-ICES/adj-env/lib/python3.9/site-packages/jax/_src/pjit.py", line 180, in _python_pjit_helper
    args_flat, _, params, in_tree, out_tree, _ = infer_params_fn(
  File "/work/07169/wgl/ls6/PHO-ICES/adj-env/lib/python3.9/site-packages/jax/_src/api.py", line 311, in infer_params
    return pjit.common_infer_params(pjit_info_args, *args, **kwargs)
  File "/work/07169/wgl/ls6/PHO-ICES/adj-env/lib/python3.9/site-packages/jax/_src/pjit.py", line 480, in common_infer_params
    jaxpr, consts, canonicalized_out_shardings_flat = _pjit_jaxpr(
  File "/work/07169/wgl/ls6/PHO-ICES/adj-env/lib/python3.9/site-packages/jax/_src/pjit.py", line 918, in _pjit_jaxpr
    jaxpr, final_consts, out_type = _create_pjit_jaxpr(
  File "/work/07169/wgl/ls6/PHO-ICES/adj-env/lib/python3.9/site-packages/jax/_src/linear_util.py", line 322, in memoized_fun
    ans = call(fun, *args)
  File "/work/07169/wgl/ls6/PHO-ICES/adj-env/lib/python3.9/site-packages/jax/_src/pjit.py", line 874, in _create_pjit_jaxpr
    jaxpr, global_out_avals, consts = pe.trace_to_jaxpr_dynamic(
  File "/work/07169/wgl/ls6/PHO-ICES/adj-env/lib/python3.9/site-packages/jax/_src/profiler.py", line 314, in wrapper
    return func(*args, **kwargs)
  File "/work/07169/wgl/ls6/PHO-ICES/adj-env/lib/python3.9/site-packages/jax/_src/interpreters/partial_eval.py", line 2049, in trace_to_jaxpr_dynamic
    jaxpr, out_avals, consts = trace_to_subjaxpr_dynamic(
  File "/work/07169/wgl/ls6/PHO-ICES/adj-env/lib/python3.9/site-packages/jax/_src/interpreters/partial_eval.py", line 2066, in trace_to_subjaxpr_dynamic
    ans = fun.call_wrapped(*in_tracers_)
  File "/work/07169/wgl/ls6/PHO-ICES/adj-env/lib/python3.9/site-packages/jax/_src/linear_util.py", line 166, in call_wrapped
    ans = self.f(*args, **dict(self.params, **kwargs))
jax._src.traceback_util.UnfilteredStackTrace: TypeError: absolute() takes 1 positional argument but 2 were given
The stack trace below excludes JAX-internal frames.
The preceding is the original exception that occurred, unmodified.
--------------------
The above exception was the direct cause of the following exception:
Traceback (most recent call last):
  File "/opt/apps/intel19/python3/3.9.7/lib/python3.9/pdb.py", line 1723, in main
    pdb._runscript(mainpyfile)
  File "/opt/apps/intel19/python3/3.9.7/lib/python3.9/pdb.py", line 1583, in _runscript
    self.run(statement)
  File "/opt/apps/intel19/python3/3.9.7/lib/python3.9/bdb.py", line 580, in run
    exec(cmd, globals, locals)
  File "<string>", line 1, in <module>
  File "/scratch/07169/wgl/adjoint/python/Main_no_matrix.py", line 200, in <module>
    v = adjointSolve(u, dt, ref_factor, params, net)
  File "/scratch/07169/wgl/adjoint/python/Main_no_matrix.py", line 60, in adjointSolve
    dJdU = grad(outFnl)(u_fine, t_fine)
  File "/scratch/07169/wgl/adjoint/python/Main_no_matrix.py", line 51, in outFnl
    loss = jnp.abs(jnp.squeeze(u[-1]), jnp.squeeze(jnp.sin(t[-1] + u[0])))
TypeError: absolute() takes 1 positional argument but 2 were given
Uncaught exception. Entering post mortem debugging
Running 'cont' or 'step' will restart the program
> /scratch/07169/wgl/adjoint/python/Main_no_matrix.py(51)outFnl()
-> loss = jnp.abs(jnp.squeeze(u[-1]), jnp.squeeze(jnp.sin(t[-1] + u[0])))
Post mortem debugger finished. The /scratch/07169/wgl/adjoint/python/Main_no_matrix.py will be restarted
> /scratch/07169/wgl/adjoint/python/Main_no_matrix.py(1)<module>()
-> import os
Program interrupted. (Use 'cont' to resume).
> /work/07169/wgl/ls6/PHO-ICES/adj-env/lib/python3.9/site-packages/jax/_src/source_info_util.py(187)current()
-> if not source_info.traceback:
Restarting /scratch/07169/wgl/adjoint/python/Main_no_matrix.py with arguments:
	
> /scratch/07169/wgl/adjoint/python/Main_no_matrix.py(1)<module>()
-> import os
/work/07169/wgl/ls6/PHO-ICES/adj-env/lib/python3.9/site-packages/plotly/matplotlylib/renderer.py:647: UserWarning:
Looks like the annotation(s) you are trying
to draw lies/lay outside the given figure size.
Therefore, the resulting Plotly figure may not be
large enough to view the full text. To adjust
the size of the figure, use the 'width' and
'height' keys in the Layout object. Alternatively,
